website/blog/2019-11-04-binding-react-navigation-5-to-ui-kitten.md

```diff
@@ -0,0 +1,429 @@
+
+[Eva Design System](https://eva.design) is a customizable Design System that is easy to adapt to your brand. It provides Mobile and Web component libraries and allows businesses to quickly create beautiful unique branding themes. The React Native realization of Eva Design System includes [UI Kitten](http://akveo.github.io/react-native-ui-kitten), React Native framework for building modern cross-platform mobile applications.
+
+The UI Kitten team started actively using React Navigation alpha and we're proud to announce the full compatibility to the new React Navigation API. In this guide, we won't consider how to implement all of the boilerplate stuff like auth screens. Instead, we will learn how to navigate between screens using Drawer, Bottom Tabs, Top Tabs, and Stack navigators to build a TODO-App. Furthermore, we'll demonstrate binding React Navigation to UI Kitten components.
+
+<img src="assets/binding-react-navigation-5-to-ui-kitten/overview.gif" height="480"/>
+
+## Overview
+
+React Navigation 5 is nothing else rather than simplifying navigation structure in your app.
+
+```js
+import { createStackNavigator } from '@react-navigation/stack';
+
+const Stack = createStackNavigator();
+
+export const AuthNavigator = (): React.ReactElement => (
+  <Stack.Navigator headerMode='none'>
+    <Stack.Screen name='Sign In' component={SignInScreen}/>
+    <Stack.Screen name='Sign Up' component={SignUpScreen}/>
+  </Stack.Navigator>
+);
+```
+
+The syntax is simple. You just import `createXNavigator` from the navigator package of your choice and use `Navigator` and `Screen` components from the value returned by this function.
+
+Unlike the previous React Navigation versions, all the screens used inside a navigator are passed as child elements with wrapping it to a `Screen` component. If you need to set up additional navigator configuration like configuration of the header, you can simply pass corresponding props directly to the `Navigator` component.
+
+## Getting started
+
+Clone the project from GitHub. It contains all the required source code for the initial setup.
+
+```bash
+git clone https://github.com/artyorsh/react-navigation-ex-demo
+```
+
+### Step 1. Authentication flow
+
+Assuming that your app users will need to authorize before getting to the home screen, we will need to create both Authentication and Home navigators. Then we're going to combine it with simple stack navigation and pick the initial screen depending on the user authorization status.
+
+Open `./src/navigators/auth.navigator.tsx` file and paste the following code:
+
+
+```js
+import React from 'react';
+import { createStackNavigator } from '@react-navigation/stack';
+import { SignInScreen, SignUpScreen, ResetPasswordScreen } from '@app-scenes/auth';
```

Comment:
I'd avoid using imports like `@app-scenes/auth` for example code and use relative imports. Otherwise it's not obvious which file we're importing just by looking at the code.